src/esp/assets/ResourceManager.cpp

```diff
@@ -1463,27 +1481,75 @@ bool ResourceManager::loadRenderAssetGeneral(const AssetInfo& info) {
-        fileImporter_->scene(fileImporter_->defaultScene());
-    if (!sceneData) {
-      ESP_ERROR() << "Cannot load scene, exiting";
+  // no default scene --- standalone OBJ/PLY files, for example
+  // take a wild guess and load the first mesh with the first material
+  if (fileImporter_->defaultScene() == -1) {
+    if ((fileImporter_->meshCount() != 0u) &&
+        meshes_.at(meshMetaData.meshIndex.first)) {
+      // addMeshToDrawables(metaData, *parent, drawables, 0, 0);
+      CORRADE_ASSERT_UNREACHABLE(
+          "Sorry, this is broken since August 2019 and needs to get fixed", {});
+    } else {
+      ESP_ERROR() << "No default scene available and no meshes found, exiting";
       return false;
     }
-    for (unsigned int sceneDataID : sceneData->children3D()) {
-      loadMeshHierarchy(*fileImporter_, meshMetaData.root, sceneDataID);
-    }
-  } else if ((fileImporter_->meshCount() != 0u) &&
-             meshes_.at(meshMetaData.meshIndex.first)) {
-    // no default scene --- standalone OBJ/PLY files, for example
-    // take a wild guess and load the first mesh with the first material
-    // addMeshToDrawables(metaData, *parent, drawables, 0, 0);
-    loadMeshHierarchy(*fileImporter_, meshMetaData.root, 0);
-  } else {
-    ESP_ERROR() << "No default scene available and no meshes found, exiting";
+  }
+
+  /* Load the scene */
+  Cr::Containers::Optional<Mn::Trade::SceneData> scene;
+  if (!(scene = fileImporter_->scene(fileImporter_->defaultScene())) ||
+      !scene->is3D() || !scene->hasField(Mn::Trade::SceneField::Parent)) {
+    ESP_ERROR() << "Cannot load scene, exiting";
     return false;
   }
+
+  // Allocate objects that are part of the hierarchy. Parent / child
+  // relationship handled at the very last because MeshTransformNode stores its
+  // children by-value in a vector inside, which would mean we'd have to move
+  // them out of here
+  Cr::Containers::Array<
+      Cr::Containers::Optional<esp::assets::MeshTransformNode>>
+      nodes{std::size_t(scene->mappingBound())};
+  for (const Cr::Containers::Pair<unsigned, int>& parent :
+       scene->parentsAsArray()) {
+    nodes[parent.first()].emplace();
+    nodes[parent.first()]->componentID = parent.first();
+  }
+
+  // Set transformations. Objects that are not part of the hierarchy are
+  // ignored, nodes that have no transformation entry retain an identity
+  // transformation.
+  for (const Cr::Containers::Pair<unsigned, Mn::Matrix4>& transformation :
+       scene->transformations3DAsArray()) {
+    if (Cr::Containers::Optional<esp::assets::MeshTransformNode>& node =
+            nodes[transformation.first()])
+      node->transformFromLocalToParent = transformation.second();
+  }
+
+  // Add mesh indices for objects that have a mesh, again ignoring nodes that
+  // are not part of the hierarchy.
+  // TODO: There can be multiple mesh assignments for one object, unfortunately
+  // MeshTransformNode is not capable of that
```

Comment:
To avoid silent breakages, I suppose it'd be good to at least print a warning here?

Reply:
- Absolutely, sounds good.

Reply:
988e04c87e46e8f081c3fa1d69219c5d0bf18007